src/bin/hq.rs

```diff
@@ -5,87 +5,124 @@ use std::{
 };
 
 use clap::{Parser, Subcommand};
-use hq_rs::{parse_filter, query, write};
 
 #[derive(Parser)]
 #[command(version, about)]
 struct Args {
-    #[clap(short, long, value_name = "FILE", help = "HCL file to read from")]
-    file: Option<String>,
-
+    // the `Read` options are duplicated here because when no command is given
+    // then the `read` command is the default and its options come from the root
     #[arg(value_name = "filter", help = "HCL filter expression")]
     filter: Option<String>,
 
+    #[clap(short, long, value_name = "FILE", help = "HCL file to read from")]
+    file: Option<String>,
+
     #[command(subcommand)]
     command: Option<Command>,
 }
 
 #[derive(Subcommand)]
 enum Command {
     #[command(about = "Read value from HCL (default)")]
-    Read,
```

Comment:
so the reason `Read` had no options was that it was just leveraging its options from the root, and that was because doing `hq read` and `hq` (no subcommand) produced identical results -- i.e. the subcommand is optional, and the default action is `read`.

we take that a level further, giving `Read` a duplicate of those options to use when the user does `hq read`. we only fall back to the root read options when no subcommand is given.